### PR TITLE
Add {line_delimiter, Char} option to inet:setopts/2 and decode_packet/3

### PR DESCRIPTION
**Pull request by saleyn**
A new {line_delimiter, Char} option allows line-oriented TCP-based protocols
to use a custom line delimiting character.  It is to be used in conjunction
with {packet, line}.

This option also works with erlang:decode_packet/3 when its first argument
is 'line'.

### CODE REVIEW

**Comment by OTP-Maintainer** [comment]
The summary line of the commit message is too long and/or ends with a "."
Make sure the whole message follows the guidelines here: https://github.com/erlang/otp/wiki/Writing-good-commit-messages.

Bad message: Add {line_delim, Char} option to inet:setopts/2 and decode_packet/3


----------------

I am a script, I am not human 

----------------

**Comment by saleyn** [comment]
Fixed commit message

**Comment by OTP-Maintainer** [comment]
Patch has passed first testings and has been assigned to be reviewed


----------------

I am a script, I am not human 

----------------

File: erts/emulator/beam/packet_parser.h [hunk 1]
```diff
@@ -104,7 +104,8 @@ int packet_get_length(enum PacketParseType htype,
 		      const char* ptr, unsigned n,  /* Bytes read so far */
 		      unsigned max_plen,      /* Packet max length, 0=no limit */
 		      unsigned trunc_len,     /* Truncate (lines) if longer, 0=no limit */
-		      int* statep);           /* Internal protocol state */
+              char     delimiter,     /* Line delimiting character */
```
Comment (proxyles) [inline erts/emulator/beam/packet_parser.h]: missing tab?

**Comment by saleyn** [comment]
Fixed the tab.

**Comment by OTP-Maintainer** [comment]
Patch has passed first testings and has been assigned to be reviewed


----------------

I am a script, I am not human 

----------------

**Comment by saleyn** [comment]
Updated the new option name to `line_delimiter`

**Comment by OTP-Maintainer** [comment]
Patch has passed first testings and has been assigned to be reviewed


----------------

I am a script, I am not human 

----------------

**Comment by IngelaAndin** [comment]
We will consider this patch for 18.1

**Comment by saleyn** [comment]
Rebased this patch on maint 18.1.  Any update on its inclusion?

**Comment by OTP-Maintainer** [comment]
Patch has passed first testings and has been assigned to be reviewed


----------------

I am a script, I am not human 

----------------

**Comment by IngelaAndin** [comment]
Yes we decided to include it, we have added some clarifications to the documentation, and it still needs
to pass our nightly builds before it can be merge, but it is getting there.

**Comment by saleyn** [comment]
Thanks for the update!

**Comment by IngelaAndin** [comment]
Merged to maint.

**Pull request closed without merging**